src/Engine/Utils.java

```diff
@@ -199,75 +201,57 @@ public static boolean isPathValid(XYCoord start, FloodFillFunctor fff, int initi
     {
       XYCoord newCoord = path.getWaypoint(i).GetCoordinates();
 
-      movePower -= fff.getTransitionCost(map, lastCoord, newCoord);
+      movePower -= fff.getTransitionCost(map, lastCoord, newCoord, team, canTravelThroughEnemies);
       lastCoord = newCoord;
       if( movePower < 0 )
       {
         return false;
       }
     }
 
-    return fff.canStandOn(map, lastCoord);
+    return fff.canStandOn(map, lastCoord, mover, includeOccupiedSpaces);
   }
 
-  /** Alias for {@link #findShortestPath(XYCoord, Unit, int, int, GameMap, boolean) findShortestPath(XYCoord, Unit, int, int, GameMap, boolean=false)} **/
-  public static GamePath findShortestPath(Unit unit, XYCoord destination, GameMap map, boolean theoretical)
-  {
-    return findShortestPath(unit, destination.xCoord, destination.yCoord, map, theoretical);
-  }
-  /** Alias for {@link #findShortestPath(XYCoord, Unit, int, int, GameMap, boolean) findShortestPath(XYCoord, Unit, int, int, GameMap, boolean=false)} **/
-  public static GamePath findShortestPath(Unit unit, int x, int y, GameMap map, boolean theoretical)
+  public static GamePath findShortestPath(Unit unit, XYCoord end, GameMap map)
   {
-    return findShortestPath(new XYCoord(unit.x, unit.y), unit, x, y, map, theoretical);
+    return findShortestPath(unit, end, map, false);
   }
-  /** Alias for {@link #findShortestPath(XYCoord, Unit, int, int, GameMap, boolean) findShortestPath(XYCoord, Unit, int, int, GameMap, boolean=false)} **/
-  public static GamePath findShortestPath(Unit unit, XYCoord destination, GameMap map)
-  {
-    return findShortestPath(unit, destination.xCoord, destination.yCoord, map, false);
-  }
-  /** Alias for {@link #findShortestPath(XYCoord, Unit, int, int, GameMap, boolean) findShortestPath(XYCoord, Unit, int, int, GameMap, boolean=false)} **/
   public static GamePath findShortestPath(Unit unit, int x, int y, GameMap map)
   {
-    return findShortestPath(unit, x, y, map, false);
+    return findShortestPath(unit, new XYCoord(x, y), map, false);
```

Comment:
Yeah, I tended to use overloads that take `XYCoord` and separate `x` and `y` values just to avoid this. Not that it's like, a huge deal, and there is an argument for standardization.

Reply:
I was annoyed by having too many overloads, and most usecases already have an XYCoord handy (and going XYCoord->X/Y is annoying because the var names are kinda long)